src/main/java/com/client/core/base/tools/entitychanger/impl/StandardEntityChanger.java

```diff
@@ -41,29 +43,38 @@ public <T> T setField(T entity, String field, Object value) {
         String finalField = fields[fields.length - 1];
         Object target = Arrays.stream(fields, 0, fields.length - 1).reduce(entity, (entityOrField, nextField) -> {
             try {
-                Method getter = entityOrField.getClass().getDeclaredMethod(getterMethodName(nextField));
-                Class<?> fieldType = getter.getReturnType();
-                Method setter = entityOrField.getClass().getDeclaredMethod(setterMethodName(nextField), fieldType);
-                Object maybeValue = getter.invoke(entityOrField);
+                PropertyDescriptor propertyDescriptor = BeanUtils.getPropertyDescriptor(entityOrField.getClass(), nextField);
+                if (propertyDescriptor == null) {
```

Comment:
Could we effectively just reuse the getValue method here?  Something like

Object target = getValue(entity, allFieldsButTheLastOne);